include/conditionals/extract-conditionals.hpp

```diff
@@ -0,0 +1,401 @@
+#include "util/conditional_restrictions.hpp"
+#include "util/for_each_pair.hpp"
+#include "util/log.hpp"
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/fusion/include/adapt_adt.hpp>
+#include <boost/geometry.hpp>
+#include <boost/geometry/index/rtree.hpp>
+#include <boost/program_options.hpp>
+#include <boost/scope_exit.hpp>
+#include <boost/spirit/include/qi.hpp>
+#include <boost/spirit/include/support_line_pos_iterator.hpp>
+
+#include <osmium/handler.hpp>
+#include <osmium/index/map/sparse_mem_array.hpp>
+#include <osmium/io/any_input.hpp>
+#include <osmium/tags/regex_filter.hpp>
+#include <osmium/visitor.hpp>
+
+#include <fstream>
+#include <unordered_map>
+#include <unordered_set>
+
+// Data types and functions for conditional restriction
+struct ConditionalRestriction
+{
+    osmium::object_id_type from;
+    osmium::object_id_type via;
+    osmium::object_id_type to;
+    std::string tag;
+    std::string value;
+    std::string condition;
+};
+
+struct LocatedConditionalRestriction
+{
+    osmium::Location location;
+    ConditionalRestriction restriction;
+};
+
+// Data types and functions for conditional speed limits
+struct ConditionalSpeedLimit
+{
+    osmium::object_id_type from;
+    osmium::object_id_type to;
+    std::string tag;
+    int value;
+    std::string condition;
+};
+
+struct LocatedConditionalSpeedLimit
+{
+    osmium::Location location;
+    ConditionalSpeedLimit speed_limit;
+};
+
+auto ParseGlobalArguments(int argc, char *argv[]);
```

Comment:
These should probably not be exposed globally, because they are just there to parse CLI parameters. This is important for the conditionals tool, but not for anything just using the functions.